internal/runtime/catalog/catalog.go

```diff
@@ -19,285 +19,315 @@ package catalog
 import (
 	"fmt"
 	"reflect"
+	goruntime "runtime"
+	"strings"
 
 	"github.com/pkg/errors"
-	"k8s.io/apimachinery/pkg/conversion"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/naming"
 	"k8s.io/kube-openapi/pkg/common"
 )
 
-// Catalog defines methods for working with rpc services and their request and response types.
+// Catalog contains mappings between Go types and the corresponding
+// GroupVersionHooks, metadata and OpenAPIDefinitions.
 type Catalog struct {
-	// scheme allow access api-machinery utilities for handling types and types conversions
+	// scheme allows access to api-machinery utilities to implement conversions of
+	// request and response types.
 	scheme *runtime.Scheme
 
-	// gvhToType and typeToGVH allows tracking relations between hook go types and corresponding gvh.
+	// gvhToType allows mapping a GroupVersionHook to the Go type.
```

Comment:
nit 
```suggestion
	// gvhToType allows mapping a GroupVersionHook to the RuntimeHook function.
```
Same in other places